### PR TITLE
Is_Binary_Search_Tree

### PR DESCRIPTION
**Pull request by Codess08**
IsBST java solution

Issue No : #1678

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Languages Used:
- Java

#### Files Added:
- isBST.java

### CODE REVIEW

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+
```
Comment (singh-shreya6) [inline Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: Please add a brief problem description as multi line comment on top.

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
+    
+    import java.util.Scanner;
```
Comment (singh-shreya6) [inline Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: 4 space indentation should be given as we go inside blocks. Please remove the extra 4 space added at start of line throughout the file.

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+	public static void main(String[] args) {
+
+	    BinaryTree bt = new BinaryTree();
+	    System.out.println(bt.isBST());
+	}
+    }
+
+   /*Sample Input :
+     4 true 2 true 1 false false true 3 false false true 6 true 5 false false true 7 false false
```
Comment (singh-shreya6) [inline Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: Can you please re-write the code to take more understandable input format. The input format like this is making it tough to visualize the tree structure you are giving as your input.

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+	public Node takeInput(Node parent, boolean ilc) {
+
+	    int cdata = scn.nextInt();
+	    Node child = new Node();
+	    child.data = cdata;
+	    this.size++;
+
+	    // left
+	    boolean hlc = scn.nextBoolean();
```
Comment (singh-shreya6) [inline Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: please give more meaningful variable names.

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java (hunk 3 above)
Comment (Bhavyaa08) [reply Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: Should I accept the tree in level order?

**Comment by somya-kapoor** [comment]
@Bhavyaa08 Are you working on it?

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java (hunk 3 above)
Comment (singh-shreya6) [reply Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: Any standard format is good.

Comment (Bhavyaa08) [reply Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: I saw the code for this question in c++ that has been accepted as a solution in this repository. In that file the tree has been hard cored, I too am doing the same for now.

**Comment by Bhavyaa08** [comment]
YEs i have made the changes requested.

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+public class BinaryTree {
+	
+    class Node {
+	int data;
+	Node left;
+	Node right;
+	    
+	public Node(int x) 
+	{
```
Comment (singh-shreya6) [inline Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: fix indentation here. Please add 4 spaces.

File: Is_Binary_Search_Tree/Is_Binary_Search_Tree.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import java.util.Scanner;
```
Comment (singh-shreya6) [inline Is_Binary_Search_Tree/Is_Binary_Search_Tree.java]: Please add brief problem description on top as multi line comment

**Comment by singh-shreya6** [comment]
Are you working on this? This PR is open from 18 days. Will close if I don't see any updates.

**Comment by Codess08** [comment]
> Are you working on this? This PR is open from 18 days. Will close if I don't see any updates.

i will update it today, please can you tell me how should i take the input of the tree?

**Pull request closed without merging**